#  Film Industry Analysis for Business Expansion 

##  **Introduction**  
As part of a company's expansion into the entertainment industry, they are exploring the creation of a new movie studio. However, with no prior experience in film production, it is critical to make data-driven decisions about the types of movies to develop.  

In this project, I will analyze movie industry data to answer key business questions:  
- *Which movie genres tend to receive the highest ratings?* 
- *Which genres have the highest return on investment (ROI)?*
- *Are certain directors more likely to deliver high foreign box office success?*  
- *Which film languages are associated with higher popularity?*  

The goal is to uncover patterns and insights that will help guide the company's decisions on what types of films to produce for maximum commercial success.


## 【1】**Data Exploration**  

### *Loading Datasets*
It is crucial we thoroughly explore each of our datasets and database tables. The datasets we are working with include:
- `Box Office Mojo`
- `IMDB`
- `Rotten Tomatoes Movies`
- `Rotten Tomatoes Critic Reviews`
- `TheMovieDB`
- `The Numbers`
- `im.db` (SQLite database)

Data exploration helps us understand the structure, quality, and characteristics of the data we are working with.  
By examining distributions, missing values, duplicates, and relationships between variables, we can identify potential issues early and plan our cleaning and transformation strategies more effectively.  
This step ensures that our later analyses are based on a solid understanding of the data, reducing the risk of errors and unexpected results.



In [30]:
# Import necessary libraries
import pandas as pd
import numpy as np
import sqlite3

In [31]:
# Establish a connection to the database
conn = sqlite3.connect('Data/im.db')

# Check available tables (optional, good for understanding structure)
pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table';", conn)


,name
0,movie_basics
1,directors
2,known_for
3,movie_akas
4,movie_ratings
5,persons
6,principals
7,writers


In [32]:
# Close connection
conn.close()

Load all Datasets into Pandas Dataframes:

In [33]:
bom_df = pd.read_csv('Data/bom.movie_gross.csv')
bom_df.head()

,title,studio,domestic_gross,foreign_gross,year
0,Toy Story 3,BV,415000000.0,652000000,2010
1,Alice in Wonderland (2010),BV,334200000.0,691300000,2010
2,Harry Potter and the Deathly Hallows Part 1,WB,296000000.0,664300000,2010
3,Inception,WB,292600000.0,535700000,2010
4,Shrek Forever After,P/DW,238700000.0,513900000,2010


In [34]:
rt_rev_df = pd.read_csv('Data/rt_critic_reviews.csv')
rt_rev_df.head()

,rotten_tomatoes_link,critic_name,top_critic,publisher_name,review_type,review_score,review_date,review_content
0,m/0814255,Andrew L. Urban,False,Urban Cinefile,Fresh,NaN,2010-02-06,A fantasy adventure that fuses Greek mythology...
1,m/0814255,Louise Keller,False,Urban Cinefile,Fresh,NaN,2010-02-06,"Uma Thurman as Medusa, the gorgon with a coiff..."
2,m/0814255,NaN,False,FILMINK (Australia),Fresh,NaN,2010-02-09,With a top-notch cast and dazzling special eff...
3,m/0814255,Ben McEachen,False,Sunday Mail (Australia),Fresh,3.5/5,2010-02-09,Whether audiences will get behind The Lightnin...
4,m/0814255,Ethan Alter,True,Hollywood Reporter,Rotten,NaN,2010-02-10,What's really lacking in The Lightning Thief i...


In [35]:
rt_df = pd.read_csv('Data/rt_movies.csv')
rt_df.head()

,rotten_tomatoes_link,movie_title,movie_info,critics_consensus,content_rating,genres,directors,authors,actors,original_release_date,...,production_company,tomatometer_status,tomatometer_rating,tomatometer_count,audience_status,audience_rating,audience_count,tomatometer_top_critics_count,tomatometer_fresh_critics_count,tomatometer_rotten_critics_count
0,m/0814255,Percy Jackson & the Olympians: The Lightning T...,"Always trouble-prone, the life of teenager Per...",Though it may seem like just another Harry Pot...,PG,"Action & Adventure, Comedy, Drama, Science Fic...",Chris Columbus,"Craig Titley, Chris Columbus, Rick Riordan","Logan Lerman, Brandon T. Jackson, Alexandra Da...",2010-02-12,...,20th Century Fox,Rotten,49.0,149.0,Spilled,53.0,254421.0,43,73,76
1,m/0878835,Please Give,Kate (Catherine Keener) and her husband Alex (...,Nicole Holofcener's newest might seem slight i...,R,Comedy,Nicole Holofcener,Nicole Holofcener,"Catherine Keener, Amanda Peet, Oliver Platt, R...",2010-04-30,...,Sony Pictures Classics,Certified-Fresh,87.0,142.0,Upright,64.0,11574.0,44,123,19
2,m/10,10,"A successful, middle-aged Hollywood songwriter...",Blake Edwards' bawdy comedy may not score a pe...,R,"Comedy, Romance",Blake Edwards,Blake Edwards,"Dudley Moore, Bo Derek, Julie Andrews, Robert ...",1979-10-05,...,Waner Bros.,Fresh,67.0,24.0,Spilled,53.0,14684.0,2,16,8
3,m/1000013-12_angry_men,12 Angry Men (Twelve Angry Men),Following the closing arguments in a murder tr...,Sidney Lumet's feature debut is a superbly wri...,NR,"Classics, Drama",Sidney Lumet,Reginald Rose,"Martin Balsam, John Fiedler, Lee J. Cobb, E.G....",1957-04-13,...,Criterion Collection,Certified-Fresh,100.0,54.0,Upright,97.0,105386.0,6,54,0
4,m/1000079-20000_leagues_under_the_sea,"20,000 Leagues Under The Sea","In 1866, Professor Pierre M. Aronnax (Paul Luk...","One of Disney's finest live-action adventures,...",G,"Action & Adventure, Drama, Kids & Family",Richard Fleischer,Earl Felton,"James Mason, Kirk Douglas, Paul Lukas, Peter L...",1954-01-01,...,Disney,Fresh,89.0,27.0,Upright,74.0,68918.0,5,24,3


In [36]:
tmdb_df = pd.read_csv('Data/tmdb.movies.csv')
tmdb_df.head()

,Unnamed: 0,genre_ids,id,original_language,original_title,popularity,release_date,title,vote_average,vote_count
0,0,"[12, 14, 10751]",12444,en,Harry Potter and the Deathly Hallows: Part 1,33.533,2010-11-19,Harry Potter and the Deathly Hallows: Part 1,7.7,10788
1,1,"[14, 12, 16, 10751]",10191,en,How to Train Your Dragon,28.734,2010-03-26,How to Train Your Dragon,7.7,7610
2,2,"[12, 28, 878]",10138,en,Iron Man 2,28.515,2010-05-07,Iron Man 2,6.8,12368
3,3,"[16, 35, 10751]",862,en,Toy Story,28.005,1995-11-22,Toy Story,7.9,10174
4,4,"[28, 878, 12]",27205,en,Inception,27.920,2010-07-16,Inception,8.3,22186


In [37]:
tn_df = pd.read_csv('Data/tn.movie_budgets.csv')
tn_df.head()

,id,release_date,movie,production_budget,domestic_gross,worldwide_gross
0,1,"Dec 18, 2009",Avatar,"$425,000,000","$760,507,625","$2,776,345,279"
1,2,"May 20, 2011",Pirates of the Caribbean: On Stranger Tides,"$410,600,000","$241,063,875","$1,045,663,875"
2,3,"Jun 7, 2019",Dark Phoenix,"$350,000,000","$42,762,350","$149,762,350"
3,4,"May 1, 2015",Avengers: Age of Ultron,"$330,600,000","$459,005,868","$1,403,013,963"
4,5,"Dec 15, 2017",Star Wars Ep. VIII: The Last Jedi,"$317,000,000","$620,181,382","$1,316,721,747"


After successfully loading all the datasets using pandas, we are now ready to dive deeper into data exploration. Given that we are working with six distinct datasets and a database, manually using `.info()`, `.describe()`, `.isna()`, and writing separate code blocks to inspect every dataframe would be highly inefficient, tedious and extremely confusing. Moreover, for the `im.db` database, exploring each table one by one through traditional code would have required an overwhelming amount of setup, making the process unnecessarily complex.

To overcome this, we identified tools that allowed us to explore our data in a far more efficient and highly visual manner.  


### *Data Exploration with Data Wrangler and SQLite Viewer*

For the initial exploration of our datasets, we utilized two powerful Visual Studio Code extensions: 

`Data Wrangler` made early-stage data exploration much faster and more intuitive by providing:
- Visual summaries of column distributions and data types.
- Instant detection of missing values, outliers, and duplicates.
- Quick sorting, filtering, and grouping operations without needing to write code.
- Profiling of datasets with automated suggestions for cleaning and transformation.

`SQLite Viewer` allowed us to efficiently browse the database structure and contents without having to write extensive SQL queries.  
With it, we could:
- Instantly preview tables, columns, and sample data.
- Navigate between tables and their relationships quickly.
- Understand data volumes, missing values, and schema designs in a highly visual and interactive way.

Using `Data Wrangler` and `SQLite Viewer` streamlined our workflow significantly.  
Instead of manually traversing large datasets and databases line-by-line in code, we were able to focus our time on identifying meaningful issues, relationships, and insights

<div style="white-space: nowrap; overflow-x: auto;">
    <img src="images/bom_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/rt_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/rt_rev_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/tmdb_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/tn_df.png" style="display: inline-block; height: 500px; margin-right: 10px;">
    <img src="images/im.db.png" style="display: inline-block; height: 500px; margin-right: 10px;">
</div>

After thorough exploration and exhaustive discussion of the datasets, we carefully evaluated the available columns. Based on importance to our project goals and quality, we decided on a final set of columns that would be most effective for a meaningful analysis:

- From `rotten_tomatoes`:
  - `movie_title`, `genres`, `tomatometer_rating`, `audience_rating`  
    → These columns help us evaluate which movie genres tend to receive the highest critical and audience ratings.

- From `tmdb`:
  - `language`, `popularity`  
    → These attributes allow us to investigate how a film's language impacts its popularity and, by extension, potential profitability.

- From `the_numbers`:
  - `movie`, `production_budget`, `worldwide_gross`  
    → These financial figures are crucial for calculating Return on Investment (ROI) and understanding broader profitability patterns across genres.

- From `im.db`:
  - Extracting and joining data across the `directors`, `persons`, and `movie_akas` tables  
    → This allows us to match each director with the correct movie title, enabling the analysis of which directors tend to deliver high foreign box office success.

Our data exploration confirmed that the columns we selected had either zero or extremely low missing values and were populated with a wide range of distinct entries, indicating that the data quality was high and free from placeholder values.
Even though these are the specific columns we are interested in, it is still necessary to properly prepare and relate the data across different datasets to build a strong foundation for answering the business questions effectively.  


### Key Notes

- **Replacement of Rotten Tomatoes Dataset:**  
  We did not use the originally provided Rotten Tomatoes datasets, as they were inadequate for our analysis.  
  The main relationship between all datasets is the `movie title`, and the provided Rotten Tomatoes tables were missing the movie title column entirely.  
  To ensure consistency and completeness, we imported a more reliable Rotten Tomatoes dataset from [Kaggle](https://www.kaggle.com/datasets/stefanoleone992/rotten-tomatoes-movies-and-critic-reviews-dataset?select=rotten_tomatoes_movies.csv).

- **Exclusion of Box Office Mojo and Rotten Tomatoes Critic Reviews:**  
  We decided not to use the Box Office Mojo dataset because its information overlapped significantly with The Numbers dataset, which had more complete and reliable financial data.  
  Similarly, the Rotten Tomatoes Critic Reviews dataset was not utilized, as the Rotten Tomatoes Movies Info dataset already provided all the necessary critical and audience rating information for our needs.


## 【2】**Data Preparation & Cleaning**  

Before conducting meaningful analysis, it is essential to properly prepare and clean the data. Even though our data exploration confirmed that we are working with high-quality datasets. Since our datasets come from a variety of origins. Preparation involves filtering relevant columns, renaming fields for consistency, handling duplications, and merging datasets where necessary.  

### Preparing Rotten Tomatoes Ratings and Genres Data

To explore the relationship between movie genres, ratings, and their potential connection to success,  
we will prepare and clean the `Rotten Tomatoes Movies` dataset. This dataset provides detailed metadata for a wide range of movies collected from the Rotten Tomatoes website, including information that will be essential for answering our business questions regarding audience perception and genre trends.

#### *Extracting Ratings and Genres Data*

We focus on extracting four key columns from the dataset, which are essential for analyzing movie success and audience perception. The columns we are interested in are:

- `movie_title`: The title of the movie as listed on Rotten Tomatoes.
- `genres`: A comma-separated list of genres assigned to each movie (e.g., Drama, Comedy, Action).
- `audience_rating` The average rating given by the audience, typically scored from 0 to 100.
- `tomatometer_rating` The percentage of positive critic reviews.

When preparing this data, we must be mindful of movies being associated with multiple genres, which requires careful parsing to ensure accurate representation. 

In [38]:
# Extracting the relevant columns
df_rt = rt_df[["movie_title","genres","audience_rating","tomatometer_rating"]].copy()

# Convert the comma-separated string into a list
df_rt['genres'] = df_rt['genres'].str.split(', ')

# Exploding the genres
rt_movies_exploded = df_rt.explode('genres').reset_index(drop=True)
rt_movies_exploded.head()

,movie_title,genres,audience_rating,tomatometer_rating
0,Percy Jackson & the Olympians: The Lightning T...,Action & Adventure,53.0,49.0
1,Percy Jackson & the Olympians: The Lightning T...,Comedy,53.0,49.0
2,Percy Jackson & the Olympians: The Lightning T...,Drama,53.0,49.0
3,Percy Jackson & the Olympians: The Lightning T...,Science Fiction & Fantasy,53.0,49.0
4,Please Give,Comedy,64.0,87.0


We **explode** the `genres` column so that each genre becomes a separate row. This transformation turns one-to-many relationships into one-to-one, making it easier to analyze the data by individual genre. After exploding the column, we should reset the index to clean up the DataFrame and ensure the data is well-organized.



In [39]:
# Comparing the number of rows for main df and the exploded df 
print(f"main: {rt_df.shape[0]} rows")
print(f"exploded: {rt_movies_exploded.shape[0]} rows")


main: 17712 rows
exploded: 39388 rows


Comparing the number of rows in the original DataFrame with the number of rows in the exploded DataFrame helps us confirm that multiple genres per movie were successfully separated into individual rows. An increase in the number of rows indicates that the transformation was successful.

In [40]:
# Seeing if the number of unique movie_title values matches with the main
print(f"exploded: {len(rt_movies_exploded['movie_title'].unique())} unique values")
print(f"main df: {len(rt_df['movie_title'].unique())} unique values")


exploded: 17106 unique values
main df: 17106 unique values


We verify that the number of unique `movie_title` entries remain consistent between the original and exploded datasets. This confirms that the process of exploding the genres did not result in any accidental duplication or loss of movies.


#### *Data Cleaning*

From our earlier data exploration, we confirmed that the columns we are focusing on - `genres`, `audience_rating`, and `tomatometer_rating` contain no duplicates or placeholders. Each column has distinct values, ensuring the data is well-structured. However, we also observed that there are between 0% to 1% of missing values in these columns. Since these missing values are minimal, we can handle them appropriately without significantly affecting the accuracy of our cleaned dataset. In this case, dropping these rows will not reduce the dataset's integrity.

In [41]:
# Checking if missing values align with data exploration findings
(rt_movies_exploded.isna().sum()/len(rt_movies_exploded)*100)

movie_title           0.000000
genres                0.048238
audience_rating       1.241495
tomatometer_rating    0.233574
dtype: float64

Our check confirms that the missing values align with our previous findings. We can now proceed with dropping the rows containing missing values.

In [42]:
# Dropping rows with missing values under 'genres', 'audience_rating', and 'tomatometer_rating'
rt_movies_exploded_cleaned = (
    rt_movies_exploded
    .dropna(subset=["genres", "audience_rating", "tomatometer_rating"])
    .reset_index(drop=True)
)

# Checking if there are any missing values left
rt_movies_exploded_cleaned.isna().sum()


movie_title           0
genres                0
audience_rating       0
tomatometer_rating    0
dtype: int64

#### *Checking for Outliers*

Outliers can skew our analysis and lead to inaccurate results. Identifying and handling them ensures the data better reflects overall trends.

We will use the Interquartile Range (IQR) method to check for outliers:
- **IQR = Q3 - Q1**
- Any data point below **Q1 - 1.5 * IQR** or above **Q3 + 1.5 * IQR** is considered an outlier.

We will apply this logic to `audience_rating` and `tomatometer_rating`



In [43]:
# Checking for outliers
def check_outliers(q1,q3,min,max):
    iqr=q3-q1
    upper_bound=q3+(1.5*iqr)
    lower_bound=q1-(1.5*iqr)
    if (min>lower_bound) & (max<upper_bound):
     return("There are no outliers")
    else:
     return("There are outliers")
 
# Checking in the audience_rating column
print(f"{check_outliers(q1=46,q3=78,min=0,max=100)} in the audience rating column")

# Checking in the tomatometer_rating column
print(f"{check_outliers(q1=40,q3=86,min=0,max=100)} in the tomatometer rating column")


There are no outliers in the audience rating column
There are no outliers in the tomatometer rating column


#### *Standardizing text columns*
Standardization helps avoid inconsistencies during grouping, searching, or filtering. (e.g., "Drama" and "drama" would otherwise be treated differently). We standardize text data by converting `movie_title` and `genres` columns to lowercase.


In [44]:
# Making sure to work on a copy to avoid the SettingWithCopyWarning
rt_movies_exploded_cleaned = rt_movies_exploded_cleaned.copy()

# Standardizing columns to lowercase
rt_movies_exploded_cleaned["genres"] = rt_movies_exploded_cleaned["genres"].str.lower()
rt_movies_exploded_cleaned["movie_title"] = rt_movies_exploded_cleaned["movie_title"].str.lower()

rt_movies_exploded_cleaned.head()

,movie_title,genres,audience_rating,tomatometer_rating
0,percy jackson & the olympians: the lightning t...,action & adventure,53.0,49.0
1,percy jackson & the olympians: the lightning t...,comedy,53.0,49.0
2,percy jackson & the olympians: the lightning t...,drama,53.0,49.0
3,percy jackson & the olympians: the lightning t...,science fiction & fantasy,53.0,49.0
4,please give,comedy,64.0,87.0


#### *Aggregating Ratings by Genre*

To better understand how different genres perform on average with audiences and critics, we grouped the cleaned dataframe by genre and calculated the mean of both the audience ratings and the tomatometer ratings. This aggregation helps us identify broader trends and preferences across movie genres, rather than focusing on individual movie performances.


In [45]:
# Grouping by genres and aggregating the mean of audience_rating and tomatometer_rating
genre_rating_df = rt_movies_exploded_cleaned.groupby('genres').agg({
    'audience_rating': 'mean',
    'tomatometer_rating': 'mean'
}).reset_index()

# Formatting the ratings to show only one decimal place
genre_rating_df['audience_rating'] = genre_rating_df['audience_rating'].round(1)
genre_rating_df['tomatometer_rating'] = genre_rating_df['tomatometer_rating'].round(1)

genre_rating_df


,genres,audience_rating,tomatometer_rating
0,action & adventure,57.9,54.2
1,animation,65.7,64.1
2,anime & manga,75.0,75.4
3,art house & international,66.5,71.0
4,classics,71.0,78.4
5,comedy,58.4,55.1
6,cult movies,55.8,60.5
7,documentary,73.3,80.9
8,drama,63.1,63.2
9,faith & spirituality,70.3,63.9


In [46]:
# Confirming no genres were lost after aggregation
len(rt_movies_exploded_cleaned['genres'].value_counts()) == len(genre_rating_df)

True

### genres and ROI data prep
#### Vincent can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

### language and popularity data prep
#### Chris can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

# Language and popularity Data Preparation
 To achieve your company's goal of identifying the most successful film types at the box office, analyzing the **language** and **popularity** columns from 'The Movie DB' will be crucial. The **language** column reveals which linguistic markets (e.g., English, Spanish) dominate box office performance, helping you prioritize production in languages with the widest audience reach or untapped potential. Meanwhile, the **popularity** column, reflecting voter engagement, highlights trends in viewer preferences—whether high popularity correlates with genres, franchises, or cultural themes that resonate globally. By cross-referencing these metrics, your company can strategically decide which language-driven markets to target and which popular themes or styles to emulate, maximizing the new studio's chances of commercial success.

## Key point
* the data in the original_language columns contains duplicates since the data may appears in different context. For example the English language may have been used in the creation of different data

# loading the dataset
the dataset being used is from The Movie DB a CSV file and since we already have it loaded we will just call th first 5 rows to understand it better

In [47]:
tmdb_df.head(5)

,Unnamed: 0,genre_ids,id,original_language,original_title,popularity,release_date,title,vote_average,vote_count
0,0,"[12, 14, 10751]",12444,en,Harry Potter and the Deathly Hallows: Part 1,33.533,2010-11-19,Harry Potter and the Deathly Hallows: Part 1,7.7,10788
1,1,"[14, 12, 16, 10751]",10191,en,How to Train Your Dragon,28.734,2010-03-26,How to Train Your Dragon,7.7,7610
2,2,"[12, 28, 878]",10138,en,Iron Man 2,28.515,2010-05-07,Iron Man 2,6.8,12368
3,3,"[16, 35, 10751]",862,en,Toy Story,28.005,1995-11-22,Toy Story,7.9,10174
4,4,"[28, 878, 12]",27205,en,Inception,27.920,2010-07-16,Inception,8.3,22186


## Step 2; Extracting the relevant columns
Since in our analysis we are focused on the relationship between the language and popularity we extract only 2 columns, original_language and popularity columns for the analysis

In [48]:
#filtering the dataset to include only relevant columns
tmdb_relevant_columns = ['original_language','popularity']

In [49]:
# calling the relevant columns from the dataset
tmdb_relevant_columns=tmdb_df[tmdb_relevant_columns]
tmdb_relevant_columns.head(5)

,original_language,popularity
0,en,33.533
1,en,28.734
2,en,28.515
3,en,28.005
4,en,27.920


## step 3 Checking the shape and description of the dataset
 Before analyzing or modeling data, examining its shape (structure) and description (statistical summary) is crucial.
 Tis will aid in explaining from what we obtained from the Data Wrangler extension
 since the data shape helps in:
+checking the data volume
### While data description aids  in:
* revealing basic statistics for numerical columns
* Exposes misssing values and data types

In [50]:
#checking the shape of the dataset
tmdb_relevant_columns.shape

(26517, 2)

In [51]:
#checking  the description of the dataset
tmdb_relevant_columns.describe(include='all')

,original_language,popularity
count,26517,26517.000000
unique,76,NaN
top,en,NaN
freq,23291,NaN
mean,NaN,3.130912
std,NaN,4.355229
min,NaN,0.600000
25%,NaN,0.600000
50%,NaN,1.374000
75%,NaN,3.694000


 * dataset contains 2 columns and 26517 rows

* from the description of the data we can see there are no missing values in both the columns since they both have the same count of 26517.
* in the original_language columns there are 76 unique data indicating we have 76 different languages that have a movie made in that language and in which en which is the English language is the one with the most movies made in it with a frequency of 23291.
* The presence of the term 'missing value ' in the mean, std, min, 25%, 50%, 75% is because the data type inthe original_language column is astring(text) and not either an interger or float

* In the popularity database the mean is (3.1309), the std (4.355), the min(0.6), the median (1.374) and the 75th percentile as 3.694

In [52]:
# Check for missing values in the 'Popularity' column
print("Missing popularity values:", tmdb_relevant_columns['popularity'].isna().sum())


Missing popularity values: 0


In [53]:
# Check for missing values in the 'original_language' column
print("Missing original_language values:", tmdb_relevant_columns['original_language'].isna().sum())

Missing original_language values: 0


# checking for duplicates within our dataset
Duplicate data can distort your results, skew statistics, and reduce the performance of your models hence important to check and remove them

In [54]:
# Check for duplicates in the dataset
print("Duplicates:", tmdb_df.duplicated().sum())

Duplicates: 0


From the dataset it was found the were no duplicates within it

# Checking for Outliers within our dataset
Outliers can have a big impact on your statistical analyses and skew the results of any hypothesis test if they are inaccurate

In [55]:
# Calculate Q1 (25th percentile) and Q3 (75th percentile)
q1 = tmdb_relevant_columns['popularity'].quantile(0.25)
q3 = tmdb_relevant_columns['popularity'].quantile(0.75)

# Compute the IQR
iqr = q3 - q1

# Determine the lower and upper bounds for outliers
lower_bound = q1 - 1.5 * iqr
upper_bound = q3 + 1.5 * iqr

# Identify outliers
outliers = tmdb_relevant_columns[
    (tmdb_relevant_columns['popularity'] < lower_bound) |
    (tmdb_relevant_columns['popularity'] > upper_bound)
]

print(f"Lower Bound: {lower_bound}")
print(f"Upper Bound: {upper_bound}")
print(f"Number of Outliers: {len(outliers)}")


Lower Bound: -4.041
Upper Bound: 8.335
Number of Outliers: 2529


 These outliers are the rows where the popularity  value is either below -4.041 or above 8.335. 

### Preparing Directors and Foreign Gross Data

We will prepare and align two datasets to explore the relationship between directors and the worldwide gross earnings of their movies. From our earlier data exploration, we have confirmed that there are no missing values in the relevant `im.db` tables and in `the Numbers movie budgets` dataset, which will allow us to proceed without needing additional imputation steps. Since these datasets originate from different sources and structures, we will need to carefully filter and merge them to ensure accuracy.  

#### *Extracting Director and Movie Title Data*

We will start by extracting a clean dataset containing each director's name and their corresponding movie title from the `im.db` database. However the `directors` table contains identifiers (IDs) that point to other tables for the necessary information. Specifically, the director's name is not stored directly in the `directors` table but rather in the `persons` table. Additionally, the movie title is stored in the `movie_akas` table, which links movies to their various titles across different regions.

To ensure we retrieve only unique director–movie pairs, we are using a `DISTINCT` query.  
During earlier exploration, we noticed several duplicate entries caused by the way data was entered into the database, so applying `DISTINCT` helps eliminate these redundancies by focusing only on original movie titles.


In [56]:
# Connect to the database
conn = sqlite3.connect('Data/im.db')

# Updated query with DISTINCT
query = '''
SELECT DISTINCT
    p.primary_name AS director_name,
    ma.title AS movie_title
FROM directors d
JOIN persons p ON d.person_id = p.person_id
JOIN movie_akas ma ON d.movie_id = ma.movie_id
WHERE ma.is_original_title = 1
'''

# Execute the query and load into a DataFrame
directors_df = pd.read_sql_query(query, conn)

# Close connection
conn.close()
directors_df.head()


,director_name,movie_title
0,Colin Trevorrow,Jurassic World
1,Andrew Stanton,John Carter
2,Marc Jampolsky,Versailles Rediscovered - The Sun King's Vanis...
3,Sam Zubrycki,Miguelito - Canto a Borinquen
4,Henning Beckhoff,Thing I Don't Get


#### *Merging Directors with Worldwide Gross Data*

We need to combine the director information with the worldwide gross earnings.  
This step ensures that each movie in our dataset is matched with both its director and its revenue performance, setting the foundation for meaningful analysis.


In [57]:
# Rename for easy merge
directors_df = directors_df.rename(columns={'movie_title': 'movie'})

# Merge the two DataFrames
merged_df = pd.merge(tn_df[['movie', 'worldwide_gross']], directors_df, on='movie', how='inner')

# Reorder columns
directors_fgross_df = merged_df[['movie', 'director_name', 'worldwide_gross']]

# Sort alphabetically by director name
directors_fgross_df = directors_fgross_df.sort_values(by='director_name').reset_index(drop=True)
directors_fgross_df


,movie,director_name,worldwide_gross
0,Akira,A.R. Murugadoss,"$19,585"
1,Bully,Aaron Alon,"$1,381,824"
2,Restless,Aaron Boltz,"$2,772,511"
3,Circle,Aaron Hann,"$10,024"
4,Vampires Suck,Aaron Seltzer,"$81,424,988"
...,...,...,...
2381,Witness,Zehao Gao,"$65,532,576"
2382,Along the Roadside,Zoran Lisinac,"$3,234"
2383,Prophecy,Zuri Rinpoche,"$22,673,340"
2384,9,Éric Tessier,"$48,559,999"


#### *Aggregating Worldwide Gross Earnings by Director*

We need to aggregate the data to account for the fact that a single director may have multiple movies. Our goal is to sum up the worldwide gross earnings for each director across all their movies, creating a table that shows the total earnings per director. This aggregation will give us a clearer picture of each director's overall financial impact.


In [58]:
# Remove commas, dollar sign and convert to float for calculation
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r',', '', regex=True)
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r'\$', '', regex=True).astype(float)

# Group by director_name and sum worldwide_gross
agg_directors_df = directors_fgross_df.groupby('director_name', as_index=False)['worldwide_gross'].sum()

# Sort by worldwide_gross in descending order
agg_directors_df = agg_directors_df.sort_values(by='worldwide_gross', ascending=False).reset_index(drop=True)

# Format numbers with commas 
agg_directors_df['worldwide_gross'] = agg_directors_df['worldwide_gross'].apply(lambda x: f"{int(x):,}")
agg_directors_df


,director_name,worldwide_gross
0,Anthony Russo,"3,902,605,502"
1,Joe Russo,"3,902,605,502"
2,Pierre Coffin,"3,713,745,331"
3,Christopher Nolan,"3,086,180,484"
4,Joss Whedon,"2,992,084,614"
...,...,...
1651,Justin Paul Miller,0
1652,Ben Wicks,0
1653,Alessandro Cassigoli,0
1654,Justin Thomas Ostensen,0


### content rating and worldwide gross data prep
#### Kevin can make edits to this markdown cell appropriately in your own branch, 
as you guys can see we have mixed both data exploration and data cleaning into one so as you are prepping the data for analysis use appropriate markdown cells where necessary to explain yourself and also do not edit the other cells without your name in it as it will raise merge conflicts when merging all our code together
so you can add as many markdown and code cells in this space before the next guys name.
also if you want to edit anything in the document make note of it and dont change it in your branch just work on the assigned code task in your branch overall fine tuning we will come to that later.
this markdown cell is just as a reminder when you start the work you can remove everything here and make appropriate subtitles and markdown cells for your code

## 【3】**Data Analysis**  

#### *1. Data Analysis*  

##### Kevin
Same as before do not touch anything outside of these cells add as many cells as you would like between here and the next name. Do not edit anythin in the document if you need to edit anything write it down and tell me about it later

##### Vincent
Same as before do not touch anything outside of these cells add as many cells as you would like between here and the next name. Do not edit anythin in the document if you need to edit anything write it down and tell me about it later

##### Chris
Same as before do not touch anything outside of these cells add as many cells as you would like between here and the next name. Do not edit anythin in the document if you need to edit anything write it down and tell me about it later

##### Data analysis for the relationship between language and popularity
Analyzing the relationship between language and popularity is crucial for making data-driven decisions in the entertainment industry, especially when launching a new studio or producing content for global audiences. By examining which languages correlate with higher popularity scores, studios can strategically allocate resources to produce films in languages that maximize reach, engagement, and profitability.


##### Analysis on which languages have the most produced movies

Determining which language has the most produced movies is crucial for a company entering the movie business and planning to establish a studio, as it provides valuable insights into market demand, audience reach, and competitive landscape. A language with a high volume of movie production—such as English—often reflects a large, engaged audience base, well-established distribution channels, and potential for international success.

To visualize the languages with the highest production volume we are going to use a barplot

##### Vihaan
Same as before do not touch anything outside of these cells add as many cells as you would like between here and the next name. Do not edit anythin in the document if you need to edit anything write it down and tell me about it later

#### *2. Hypothesis Testing*  

## 【4】**Business Recommendations**  

## 【5】**Limitations of Our Analysis**  